app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from functools import wraps
+from typing import Any, Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_dict = {}
+
+    @wraps(func)
+    def inner(*args) -> Any:
```

Comment:
What if the function takes key arguments?
Also what if decorator is applied to two different functions that take the same arguments?